src/hotspot/cpu/aarch64/aarch64_neon.ad

```diff
@@ -3702,3 +3996,2150 @@ instruct vabd2D(vecX dst, vecX src1, vecX src2)
+  format %{ "ssra    $dst, $src, $shift\t# vector (2S)" %}
+  ins_encode %{
+    __ ssra(as_FloatRegister($dst$$reg), __ T2S,
+            as_FloatRegister($src$$reg),
+            (int)$shift$$constant);
+  %}
+  ins_pipe(vshift64_imm);
+%}
+
+instruct vsraa4I_imm(vecX dst, vecX src, immI shift) %{
+  predicate(n->as_Vector()->length() == 4);
+  match(Set dst (AddVI dst (RShiftVI src (RShiftCntV shift))));
+  ins_cost(INSN_COST);
+  format %{ "ssra    $dst, $src, $shift\t# vector (4S)" %}
+  ins_encode %{
+    __ ssra(as_FloatRegister($dst$$reg), __ T4S,
+            as_FloatRegister($src$$reg),
+            (int)$shift$$constant);
+  %}
+  ins_pipe(vshift128_imm);
+%}
+
+instruct vsraa2L_imm(vecX dst, vecX src, immI shift) %{
+  predicate(n->as_Vector()->length() == 2);
+  match(Set dst (AddVL dst (RShiftVL src (RShiftCntV shift))));
+  ins_cost(INSN_COST);
+  format %{ "ssra    $dst, $src, $shift\t# vector (2D)" %}
+  ins_encode %{
+    __ ssra(as_FloatRegister($dst$$reg), __ T2D,
+            as_FloatRegister($src$$reg),
+            (int)$shift$$constant);
+  %}
+  ins_pipe(vshift128_imm);
+%}
+
+instruct vsrla8B_imm(vecD dst, vecD src, immI shift) %{
+  predicate(n->as_Vector()->length() == 4 ||
+            n->as_Vector()->length() == 8);
+  match(Set dst (AddVB dst (URShiftVB src (RShiftCntV shift))));
+  ins_cost(INSN_COST);
+  format %{ "usra    $dst, $src, $shift\t# vector (8B)" %}
+  ins_encode %{
+    int sh = (int)$shift$$constant;
+    if (sh >= 8) {
+      __ eor(as_FloatRegister($src$$reg), __ T8B,
+             as_FloatRegister($src$$reg),
+             as_FloatRegister($src$$reg));
+    } else {
+      __ usra(as_FloatRegister($dst$$reg), __ T8B,
+             as_FloatRegister($src$$reg), sh);
+    }
+  %}
+  ins_pipe(vshift64_imm);
+%}
+
+instruct vsrla16B_imm(vecX dst, vecX src, immI shift) %{
+  predicate(n->as_Vector()->length() == 16);
+  match(Set dst (AddVB dst (URShiftVB src (RShiftCntV shift))));
+  ins_cost(INSN_COST);
+  format %{ "usra    $dst, $src, $shift\t# vector (16B)" %}
+  ins_encode %{
+    int sh = (int)$shift$$constant;
+    if (sh >= 8) {
+      __ eor(as_FloatRegister($src$$reg), __ T16B,
+             as_FloatRegister($src$$reg),
+             as_FloatRegister($src$$reg));
+    } else {
+      __ usra(as_FloatRegister($dst$$reg), __ T16B,
+             as_FloatRegister($src$$reg), sh);
+    }
+  %}
+  ins_pipe(vshift128_imm);
+%}
+
+instruct vsrla4S_imm(vecD dst, vecD src, immI shift) %{
+  predicate(n->as_Vector()->length() == 2 ||
+            n->as_Vector()->length() == 4);
+  match(Set dst (AddVS dst (URShiftVS src (RShiftCntV shift))));
+  ins_cost(INSN_COST);
+  format %{ "usra    $dst, $src, $shift\t# vector (4H)" %}
+  ins_encode %{
+    int sh = (int)$shift$$constant;
+    if (sh >= 16) {
+      __ eor(as_FloatRegister($src$$reg), __ T8B,
+             as_FloatRegister($src$$reg),
+             as_FloatRegister($src$$reg));
+    } else {
+      __ usra(as_FloatRegister($dst$$reg), __ T4H,
```

Comment:
I see you have fixed this typo, from ushr to usra. I presume original version generates wrong code and produces wrong results for specific case? If so, do you think it deserves a separate fix, e.g. for jdk16?